.github/workflows/test.yml

```diff
@@ -3,23 +3,30 @@ name: Build and Test
 
 on:
   push:
+    branches:
+      - main
   pull_request:
-  repository_dispatch:
-  schedule:
-     - cron: '05 5 1 * *' # <https://crontab.guru/#05_5_1_*_*> - "At 05:05 on day-of-month 1"  
+#  repository_dispatch:
+#  schedule:
+#     - cron: '05 5 1 * *' # <https://crontab.guru/#05_5_1_*_*> - "At 05:05 on day-of-month 1"
 
 
 jobs:
 
   build-and-test:
-    runs-on: windows-latest 
-    
+    runs-on:
+#      - windows-latest
+      - ubuntu-latest
```

Comment:
depending on how long these take, we might want to move them to our ephemeral runners. lets leave them here for now though.